### PR TITLE
[GCU] Update RDMA Platform Validator

### PR DESCRIPTION
**Pull request by isabelmsft**
#### What I did
-Fix RDMA platform validator PR to consider two edge cases:
1) Allow empty string value "" when the operation is remove in jsonpatch. Previously, empty string was not an allowed value, so these remove operations were incorrectly blocked. This is fixed by adding the empty string "" in the platform validator conf file for RDMA tables where remove operation is supported (only PFC_WD). 
2) Ensure proper string parsing for an edge case jsonpatch path and value combination. The edge case combination is: the path includes contains a variable digit (such as /PFC_WD/Ethernet64 where 64 is the variable digit) and value is a dict. In this scenario, we need to clean the path and value combination differently so that the relevant value is properly extracted without an extra "/". 
 
-add support for Mellanox spc2, spc3 asic
#### How I did it
Update GCU x RDMA platform validator functionality
#### How to verify it
E2E testing, test_add_rack passes
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.18, pytest-4.6.11, py-1.11.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /data/sonic-mgmt-int/tests, inifile: pytest.ini
plugins: metadata-1.11.0, allure-pytest-2.8.22, html-1.22.1, xdist-1.28.0, forked-1.3.0, celery-4.4.7, repeat-0.9.1, ansible-2.2.4
['conf-name', 'group-name', 'topo', 'ptf_image_name', 'ptf', 'ptf_ip', 'ptf_ipv6', 'server', 'vm_base', 'dut', 'inv_name', 'auto_recover', 'comment']
Finished testbed info generating.
collected 1 item

configlet/test_add_rack.py::test_add_rack PASSED                         [100%]
```

test_pfcwd_status
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.18, pytest-4.6.11, py-1.11.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /data/sonic-mgmt-int/tests, inifile: pytest.ini
plugins: metadata-1.11.0, allure-pytest-2.8.22, html-1.22.1, xdist-1.28.0, forked-1.3.0, celery-4.4.7, repeat-0.9.1, ansible-2.2.4
['conf-name', 'group-name', 'topo', 'ptf_image_name', 'ptf', 'ptf_ip', 'ptf_ipv6', 'server', 'vm_base', 'dut', 'inv_name', 'auto_recover', 'comment']
Finished testbed info generating.
collected 4 items

generic_config_updater/test_pfcwd_status.py::test_stop_pfcwd[single] PASSED [ 25%]
generic_config_updater/test_pfcwd_status.py::test_stop_pfcwd[all] PASSED [ 50%]
generic_config_updater/test_pfcwd_status.py::test_start_pfcwd[single] PASSED [ 75%]
generic_config_updater/test_pfcwd_status.py::test_start_pfcwd[all] PASSED [100%]

---------- generated xml file: /data/sonic-mgmt-int/tests/logs/tr.xml ----------
```

### CODE REVIEW

File: generic_config_updater/field_operation_validators.py [hunk 1]
```diff
@@ -75,7 +79,7 @@ def _get_fields_in_patch():
 
         if 'value' in patch_element.keys() and isinstance(patch_element['value'], dict):
             for key in patch_element['value']:
-                cleaned_fields.append(cleaned_field+ '/' + key)
+                cleaned_fields.append(cleaned_field + '/' + key) if len(cleaned_field) > 0 else cleaned_fields.append(key)
```
Comment (qiluo-msft) [inline generic_config_updater/field_operation_validators.py:82]: >if [](http://example.com/codeflow?start=65&length=2)

It's better to use if-else block. Single line is too long.

Comment (isabelmsft) [reply generic_config_updater/field_operation_validators.py]: Changed to if-else block

**Review by qiluo-msft — APPROVED** [review]

**Review by wen587 — APPROVED** [review]

File: generic_config_updater/gcu_field_operation_validators.conf.json [hunk 2]
```diff
@@ -123,6 +134,7 @@
                         "operations": ["replace"],
                         "platforms": {
                             "spc1": "20181100",
```
Comment (neethajohn) [inline generic_config_updater/gcu_field_operation_validators.conf.json:136]: spc2 missed here

**Review by neethajohn — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by StormLiangMS** [comment]
@isabelmsft ADO?

**Pull request merged**